src/NuGet.Core/NuGet.Packaging/Rules/UpholdBuildConventionRule.cs

```diff
@@ -15,82 +16,122 @@ namespace NuGet.Packaging.Rules
     internal class UpholdBuildConventionRule : IPackageRule
     {
         private static ManagedCodeConventions ManagedCodeConventions = new ManagedCodeConventions(new RuntimeGraph());
-        string[] _folders = new string[]
+        private static readonly IReadOnlyList<string> BuildFolders = new string[]
         {
             PackagingConstants.Folders.Build,
             PackagingConstants.Folders.BuildTransitive,
-            PackagingConstants.Folders.BuildCrossTargeting
+            PackagingConstants.Folders.BuildCrossTargeting,
+            "buildMultiTargeting"
         };
 
         public string MessageFormat => AnalysisResources.BuildConventionIsViolatedWarning;
 
         public IEnumerable<PackagingLogMessage> Validate(PackageArchiveReader builder)
         {
-            var warnings = new List<PackagingLogMessage>();
-            foreach (var folder in _folders)
+            var packageId = builder.NuspecReader.GetId();
+            var filesInPackage = builder.GetFiles();
+
+            var violations = IdentifyViolators(filesInPackage, packageId);
+            var warning = GenerateWarning(violations);
+            return warning == null
+                ? Array.Empty<PackagingLogMessage>()
+                : new[] { warning };
+        }
+
+        internal PackagingLogMessage GenerateWarning(ICollection<ConventionViolator> conventionViolators)
```

Comment:
Why generate one message instead of multiple messages?

Reply:
It was a decision made when the validation rule was initially created. Rather than spamming the customer with many messages about fundamentally the same thing (props/targets filename not matching NuGet's conventions), have 1 message, but provide a specific list about exactly which TFMs & files are missing, so the customer doesn't have to guess.